include/segment_symbols.h

```diff
@@ -76,147 +76,16 @@ DECLARE_OVERLAY_SEGMENT(daytelop)
-DECLARE_OVERLAY_SEGMENT(En_Encount2)
-DECLARE_OVERLAY_SEGMENT(En_Fire_Rock)
-DECLARE_OVERLAY_SEGMENT(Bg_Ctower_Rot)
-DECLARE_OVERLAY_SEGMENT(Mir_Ray)
-DECLARE_OVERLAY_SEGMENT(En_Sb)
-DECLARE_OVERLAY_SEGMENT(En_Bigslime)
-DECLARE_OVERLAY_SEGMENT(En_Karebaba)
-DECLARE_OVERLAY_SEGMENT(En_In)
-DECLARE_OVERLAY_SEGMENT(En_Bom_Chu)
-DECLARE_OVERLAY_SEGMENT(En_Horse_Game_Check)
-DECLARE_OVERLAY_SEGMENT(En_Rr)
-DECLARE_OVERLAY_SEGMENT(En_Fr)
-DECLARE_OVERLAY_SEGMENT(En_Fishing)
-DECLARE_OVERLAY_SEGMENT(Obj_Oshihiki)
-DECLARE_OVERLAY_SEGMENT(Eff_Dust)
-DECLARE_OVERLAY_SEGMENT(Bg_Umajump)
-DECLARE_OVERLAY_SEGMENT(En_Insect)
-DECLARE_OVERLAY_SEGMENT(En_Butte)
-DECLARE_OVERLAY_SEGMENT(En_Fish)
-DECLARE_OVERLAY_SEGMENT(Item_Etcetera)
-DECLARE_OVERLAY_SEGMENT(Arrow_Fire)
-DECLARE_OVERLAY_SEGMENT(Arrow_Ice)
-DECLARE_OVERLAY_SEGMENT(Arrow_Light)
-DECLARE_OVERLAY_SEGMENT(Obj_Kibako)
-DECLARE_OVERLAY_SEGMENT(Obj_Tsubo)
-DECLARE_OVERLAY_SEGMENT(En_Ik)
-DECLARE_OVERLAY_SEGMENT(Demo_Shd)
-DECLARE_OVERLAY_SEGMENT(En_Dns)
-DECLARE_OVERLAY_SEGMENT(Elf_Msg)
-DECLARE_OVERLAY_SEGMENT(En_Honotrap)
-DECLARE_OVERLAY_SEGMENT(En_Tubo_Trap)
-DECLARE_OVERLAY_SEGMENT(Obj_Ice_Poly)
-DECLARE_OVERLAY_SEGMENT(En_Fz)
-DECLARE_OVERLAY_SEGMENT(En_Kusa)
-DECLARE_OVERLAY_SEGMENT(Obj_Bean)
-DECLARE_OVERLAY_SEGMENT(Obj_Bombiwa)
-DECLARE_OVERLAY_SEGMENT(Obj_Switch)
-DECLARE_OVERLAY_SEGMENT(Obj_Lift)
-DECLARE_OVERLAY_SEGMENT(Obj_Hsblock)
-DECLARE_OVERLAY_SEGMENT(En_Okarina_Tag)
-DECLARE_OVERLAY_SEGMENT(En_Goroiwa)
-DECLARE_OVERLAY_SEGMENT(En_Daiku)
-DECLARE_OVERLAY_SEGMENT(En_Nwc)
-DECLARE_OVERLAY_SEGMENT(Item_Inbox)
-DECLARE_OVERLAY_SEGMENT(En_Ge1)
-DECLARE_OVERLAY_SEGMENT(Obj_Blockstop)
-DECLARE_OVERLAY_SEGMENT(En_Sda)
-DECLARE_OVERLAY_SEGMENT(En_Clear_Tag)
-DECLARE_OVERLAY_SEGMENT(En_Gm)
-DECLARE_OVERLAY_SEGMENT(En_Ms)
-DECLARE_OVERLAY_SEGMENT(En_Hs)
-DECLARE_OVERLAY_SEGMENT(Bg_Ingate)
-DECLARE_OVERLAY_SEGMENT(En_Kanban)
-DECLARE_OVERLAY_SEGMENT(En_Attack_Niw)
-DECLARE_OVERLAY_SEGMENT(En_Mk)
-DECLARE_OVERLAY_SEGMENT(En_Owl)
-DECLARE_OVERLAY_SEGMENT(En_Ishi)
-DECLARE_OVERLAY_SEGMENT(Obj_Hana)
-DECLARE_OVERLAY_SEGMENT(Obj_Lightswitch)
-DECLARE_OVERLAY_SEGMENT(Obj_Mure2)
-DECLARE_OVERLAY_SEGMENT(En_Fu)
-DECLARE_OVERLAY_SEGMENT(En_Stream)
-DECLARE_OVERLAY_SEGMENT(En_Mm)
-DECLARE_OVERLAY_SEGMENT(En_Weather_Tag)
-DECLARE_OVERLAY_SEGMENT(En_Ani)
-DECLARE_OVERLAY_SEGMENT(En_Js)
-DECLARE_OVERLAY_SEGMENT(En_Okarina_Effect)
-DECLARE_OVERLAY_SEGMENT(En_Mag)
-DECLARE_OVERLAY_SEGMENT(Elf_Msg2)
-DECLARE_OVERLAY_SEGMENT(Bg_F40_Swlift)
-DECLARE_OVERLAY_SEGMENT(En_Kakasi)
-DECLARE_OVERLAY_SEGMENT(Obj_Makeoshihiki)
-DECLARE_OVERLAY_SEGMENT(Oceff_Spot)
-DECLARE_OVERLAY_SEGMENT(En_Torch)
-DECLARE_OVERLAY_SEGMENT(Shot_Sun)
-DECLARE_OVERLAY_SEGMENT(Obj_Roomtimer)
-DECLARE_OVERLAY_SEGMENT(En_Ssh)
-DECLARE_OVERLAY_SEGMENT(Oceff_Wipe)
+#define DEFINE_ACTOR(name, _1, _2, _3) DECLARE_OVERLAY_SEGMENT(name)
+#define DEFINE_ACTOR_INTERNAL(_0, _1, _2, _3)
+#define DEFINE_ACTOR_UNSET(_0)
```

Comment:
I don't really see the benefit of not giving the arguments proper names here.

Reply:
I just followed what OoT did here but I can name those arguments if it is preferred

Reply:
I think the point of not giving proper arguments is that those arguments aren't relevant/are completely unused to the `#define`/`#undef` in this particular case, and should be used in a different define. Having all the defines in one place is the purose of the comments at the `actor_table.h` file. I think it's clearer to see which of the many arguments are used for this particular define, so I like the way it was previously

Reply:
We don't rename function arguments if they're not used, I think it's better to see what's omitted.

Reply:
This is a macro though, which is slightly different to a function. In particular, it's the same macro that's redefined multiple times, which different arguments used in different defines. So having macro arguments stand out helps differentiate the multi-use defines imo. Could be open to more opinions though

Reply:
I agree with engineer here. Its a special case in that the same macro is used in many different contexts for different purposes. Its not at all the same as function arguments.

It helps you understand as the reader which arguments are relevant for that particular use case of the different arguments the table provides. It is very helpful to have

Reply:
I'll unname them then

Reply:
I still don't think this is less confusing, it reminds me of `inline`: https://stackoverflow.com/a/216546 .

Reply:
if you take issue with the underscore syntax specifically, we can possibly find something that works better for everyone, but I do hope our reasoning for wanting the irrelevant arguments unnamed makes sense. 

Its not a typical macro which exists to act on a defined set of inputs in the same way every time. Its a pretty non-standard thing we're doing here, that has a table defining *all* of the relevant data on a larger scale, and then context dependent macros use only specific parts of the data from that table when relevant.

Knowing at a glance which arguments are actually being used in the different re-definitions of the macro is useful information to have.

Reply:
I'm with Elliptic here. I think it's awkward to unname things.

Generally in MM we use the type to convey that something is unknown/unknownable. Granted in a macro we don't have that.

I would heavily prefer we give it a name to tell people what information is expected. We can append it with `unused` if we want to signify the information that Engineer is concerned about, which I would be fine with.

Reply:
This is not a normal macro. The precedent from your normal way of handing "unused" things doesnt really apply here. This macro specifically exists to have only some arguments be relevant in different contexts. It is ok for it to be different from the normal way of handling unused args in this case.

> I would heavily prefer we give it a name to tell people what information is expected

Thats the thing, you dont use this macro normally in the sense that you provide the macro arguments when you want to use it. You include the data from the header in the tables folder, and the naming of the arguments tells you which data FROM the table gets used. You arent passing the data in yourself.

ie you will never write `DEFINE_ACTOR(actorName, id, alloctype, string)` in the code yourself. You dont need to hover the macro and see which args it needs. You dont use this macro directly in code yourself.

Reply:
My two cents, I agree that the names would be best omitted. They are already named and well documented in `actor_table.h`, so would prefer just the numbers to show that they are irrelevant to the current macro which will be defined multiple times. Appending `unused` to would just clutter it imo.

Reply:
My preferred choice would be to precede the unused fields with underscores but keep the actual names.

Reply:
I went with the "keep names with an underscore prefix" option, I hope it is a good middle ground for everybody